common/models/user.js

```diff
@@ -591,7 +591,11 @@ module.exports = function(User) {
             query && query.redirect !== undefined ? query.redirect :
             undefined;
 
-          ctx.res.redirect(redirectUrl);
+          if (redirectUrl !== undefined) {
+            ctx.res.location(redirectUrl);
+            ctx.res.status(302);
+          }
+          next();
         } else {
           next(new Error('transport unsupported'));
         }
```

Comment:
While you are changing this part, please rework the implementation as follows:

 1. Use `ctx.args.redirectUrl` to obtain the parameter value, don't use `ctx.req` directly.
 2. Don't throw `Error("transport unsupported")` when `redirectUrl` is not set